hooks/useIsMobile.ts

```diff
@@ -0,0 +1,20 @@
+import { useEffect, useState } from 'react'
+
+export const useIsMobile = () => {
+    const [isMobile, setIsMobile] = useState<boolean>(false)
+
+    useEffect(() => {
+        const handleIsMobile = () => {
```

Comment:
This could be refactor:

`const handleIsMobile = () => setIsMobile(window.innerWidth <= 1024)`